### DISCUSSION TITLE
netbox.netbox.nb_lookup not found

### DISCUSSION THREAD

**Discussion by ayoemman** [netbox-community/ansible_modules#1313] [General]
I am trying to setup Redhat ansible automation platform with netbox, but when I try a basic playbook to test functionality:

- name: Playbook to interact with NetBox
  hosts: localhost
  gather_facts: no
  tasks:
    # query a list of devices
    - name: Obtain list of devices from NetBox
      debug:
        msg: >
          "Device {{ item.value.display_name }} (ID: {{ item.key }}) was
          manufactured by {{ item.value.device_type.manufacturer.name }}"
      loop: "{{ query('netbox.netbox.nb_lookup', 'devices',
                      api_endpoint='https://removed/',
                      token='removed') }}"

I keep getting the following error:

PLAY [Playbook to interact with NetBox] ****************************************07:35:16
TASK [Obtain list of devices from NetBox] **************************************07:35:16
fatal: [localhost]: FAILED! => {"msg": "lookup plugin (netbox.netbox.nb_lookup) not found"}
PLAY RECAP *********************************************************************07:35:16
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

All paths are correctly referenced, netbox.netbox is installed with the required python modules.
Has anyone come across this issue before? You assistance will be of great help. Thank you!

**Comment by richbibby** [comment]
did you install inside a python virtual environment? is it activated? maybe this [guide](https://github.com/netboxlabs/netbox-learning/tree/develop/netbox-ansible-collection-quick-start) is useful (see use case 3)

**Reply by ayoemman** [reply]
@richbibby Thank you for responding. I tried the method in the guide you provided earlier, and it worked fine in the terminal. I cannot do it to make it work inside the RedHat Ansible Automation Platform GUI. I hope you understand what I mean.

**Reply by richbibby** [reply]
I see what you mean. Not an expert on AAP but I would imagine this means the collection is not installed in the execution environment that you are running this playbook in. Do other playbooks that use the collection work for you on AAP?

**Reply by ayoemman** [reply]
Thank you so much. Your contribution pointed me in the right direction. I had to build an image for the EE. It works now.
